contents/zeromq/zeromq.md

```diff
@@ -0,0 +1,97 @@
+
+ZeroMQ looks like an embeddable networking library (read: easy to use) but acts as a concurrency framework (read: high performance). Using Python's built-in socket library as an example, if we are to write a simple TCP server in <tooltip content="It's just a language of my choice. ZeroMQ provides bindings for most mainstream programming languages, which are listed [here](http://wiki.zeromq.org/bindings:_start)">Python</tooltip>, it looks like this:
+
+```python
+import socket
+
+s = socket.socket() # Create the socket
+s.bind(('', 12345)) # Bind socket to a port
+s.listen(5) # Listen on the socket
+  
+# Communicate using the socket
+while True:
+   c, addr = s.accept() # Receive input
+   c.send('Connected. Now closing connection.') # Send response
+   c.close()
+```
+
+Implementing the same functionality in ZeroMQ, it looks like this:
+
+```python
+import time
+import zmq
+
+context = zmq.Context()
+socket = context.socket(zmq.REP) # Create the socket
+socket.bind("tcp://*:12345") # Bind socket to a port
+
+# Communicate using the socket
+while True:
+    message = socket.recv() # Receive input
+    socket.send(b"Connected. Now closing connection.") # Send response
+```
+
+The similarity between the two is rather obvious - as long as you have experience with socket programming, using ZeroMQ to make communication between your concurrent processes possible and efficient should be no problem at all.
+
+### Multi-Transport
+
+Unlike system calls, ZeroMQ supports various transports like in-process, inter-process, TCP, and multicast. And you don't need to change your code to switch underlying transport as ZeroMQ abstracted the details away, leaving you a reusable messaging layer. All you need to do is to update the definition (i.e. `context.socket(...)` and `socket.bind(...)`) of the socket and you're ready to scale your application from multi-core concurrency to a cluster of machines.
+
+### High Performance
+
+Unlike networking libraries, ZeroMQ performs well enough to "be the fabric for clustered products". To give you a rough idea of how lightweight ZeroMQ is, the C++ source code for the ZeroMQ core engine [libzmq](https://github.com/zeromq/libzmq/tree/master/src) is only 1.5 megabytes. Moreover, in the [benchmark](http://wiki.zeromq.org/area:results) for ZeroMQ, the latency is measured in microseconds and the throughput is measured in million messages per second. Even CERN [approved](http://cds.cern.ch/record/1391410/files/CERN-ATS-2011-196.pdf?version=1) the performance of ZeroMQ in the setting of operating accelerators.
```

Comment:
`Unlike networking libraries,`: mention some of those networking libraries you are comparing to? Can be a tooltip.